Add a "Learn" page

jvns opened the pull request:
We talked about making a "Learn" page in https://github.com/git/git-scm.com/issues/2046 and folks seemed enthusiastic, so I took a stab at making a basic page. You can see it here: https://jvns.github.io/git-scm.com/learn. The idea is now that "Reference" links directly to the reference documentation, and that the old `/doc/` page will redirect to the reference documentation.

I used Font Awesome icons (GPL compatible) because there was no "Learn" or "Videos" icon, and I wanted the icons to match. I also moved a couple of things to use flexbox/grid honestly because I have never learned how `float: left` works and so it was much easier for me to make the changes I needed to make by using flexbox or grid. Happy to come up with a different approach if folks are concerned about [grid availability in browsers](https://caniuse.com/css-grid). 

I removed links to a couple of cheat sheets on the "Docs" page. Potentially worth adding those back later if/when "External Links" gets revisited. (I'm collecting some [resources here](https://gist.github.com/jvns/1ae108f2dbe5ec0e1c42bbc6b3770c2f))

Problems I still need to resolve:

- [x] Fix dark mode on both homepage and learn page
- [x] Need to rebuild book so that it appears in the "Learn" section instead of "Documentation", as well as the docs to set up the redirect. And test that the `/doc => /docs/` redirect actually works.
- [x] <s>I haven't thought too hard about the page's copy yet. I think it would be nice to describe the videos and the book a tiny bit more.</s> (I think I'm out of ideas here right now)
- [x] The cheat sheet isn't actually there yet, need to rebase once #2049 is merged

This closes https://github.com/git/git-scm.com/issues/2055


<img width="544" height="593" alt="image" src="https://github.com/user-attachments/assets/61d9280a-bb78-49f8-bb8f-b88cf068bbe2" />
<img width="517" height="278" alt="image" src="https://github.com/user-attachments/assets/c6bc618e-bb6b-4deb-83fb-d619061403e5" />

jvns committed:
Link to Learn page on the homepage

Also use flexbox instead of `float: left` to lay out everything in
a grid; it looks like this CSS was written before flexbox existed and
flexbox is simpler.

jvns committed:
Add Learn section to sidebar

Move videos, Pro Git book, and external links to Learn section.

Will require the book CI to rerun to update the section of the site the
book is in.

jvns committed:
Add Learn page

- Add a new videos summary shortcode to make the videos more compact
- Add a new `.callout.icon` class to give the page a more unified look
- Add font awesome icons

jvns committed:
Remove /doc landing page

This makes the reference docs immediately available for the sidebar.

Make /doc an alias for /docs (the reference landing page), to
avoid broken links.

jvns committed:
Use Font Awesome for homepage icons

There was no "Learn" or "Videos" icon, and I wanted the icons to match,
so we need new icons! There are GPL compatible according to their
website.

Integrated these icons as SVGs (just copied and pasted from their site)
to keep file sizes down, the main Font Awesome CSS file is pretty big.

This should make it easier to update the icons in the future. Switch
from using `float:` to `display: grid` to display the icons.

To1ne commented:
Amazing!

jvns committed:
Link images on Learn page and fix dark mode

jvns committed:
Fix dark mode in front nav

jvns committed:
Remove unused ref-manual CSS

Commit:
Update manual pages (manually forced rebuild)

Updated via the `update-git-version-and-manual-pages.yml` GitHub workflow.

To1ne commented:
@jvns The big orange button to "Read the book" is a bit distracting. It pulls in too much attention making you think you need to click it before you can continue to the next step (next box) in the "Learn" process.

To1ne commented:
I wonder if GUIs also belong under Learn? I think they are too far hidden under Download.

To1ne commented:
I like you've put "Learn" front and center as well:

| Before | After |
|-|-|
| <img width="1266" height="576" alt="image" src="https://github.com/user-attachments/assets/23eaa631-ada9-485f-b65e-c09751f75a83" /> | <img width="1294" height="714" alt="image" src="https://github.com/user-attachments/assets/15353962-be15-4ff3-ae99-889a8f68499a" /> |

I'm not fully sold on the Font Awesome icons, they don't really fit the style I think. But since we don't have access to a graphic designer at the moment, I think that's the best we've got.

I love the squircles though!

jvns commented:
Can you say what it is about the Font Awesome icons? I'm wondering if there are some easy tweaks to make it feel better. For me I think "book" icon and the "download" icons are my least favourites, I liked how the original book icon had pages. 

The squircles were actually an accident -- what happened was that the circles were cutting off part of the image, so I made it a squircle instead to fix that. I'm glad you think it works!

jvns committed:
learn page: remove 'Read the book!'

jvns commented:
Removed the "Read the book!" button, I don't think it's necessary since that div is already highlighted.

I tried adding "GUI Clients" under "Learn" but it didn't feel right to me. Maybe a separate Tools section, kind of like https://www.rust-lang.org/tools? There are other great Git tools out there other than GUI clients that could be featured, like https://pre-commit.com/ or [git-absorb](https://github.com/tummychow/git-absorb)

To1ne commented:
> Can you say what it is about the Font Awesome icons?

@jvns I'm purely talking from a personal taste. But I think they are too roundish. As far as I know they are designed to work at smaller size, and upsizing them makes them also a little too bold for my taste (compare the speech bubbles for example, line width is too thick now). And the corners are too round for my liking.

> Removed the "Read the book!" button, I don't think it's necessary since that div is already highlighted.

:purple_heart: 

> I tried adding "GUI Clients" under "Learn" but it didn't feel right to me. Maybe a separate Tools section, kind of like https://www.rust-lang.org/tools?

Sounds good. But can happen separately. Shall I file and issue?

>  There are other great Git tools out there other than GUI clients that could be featured, like https://pre-commit.com/ or [git-absorb](https://github.com/tummychow/git-absorb)

And I think TUI tools (like lazygit) and editor extensions are better separated too.

To1ne commented:
> Sounds good. But can happen separately. Shall I file and issue?

#2070 done!

jvns committed:
Remove duplicate cheat sheet entry from sidebar

jvns committed:
Move cheat sheet to 'learn' section

jvns commented:
I've rebased this and fixed the issues I saw. I tried to look for icon packs briefly but didn't find anything, but hopefully it's easy to change the icons later if we find something better. Need to wait for CI to make sure that no links are broken.

To1ne commented:
> I tried to look for icon packs briefly but didn't find anything, but hopefully it's easy to change the icons later if we find something better.

Yeah, don't worry about them right now. I've opened https://github.com/git/git-scm.com/issues/2077

dscho reviewed (approved):
I like the new structure!

jvns commented:
Will resolve the merge conflicts with the Persian translation and merge.

dscho commented:
> Will resolve the merge conflicts with the Persian translation and merge.

Oh sorry! I did not realize that merging #2091 would cause merge conflicts here... 😞

Commit:
book: update gr

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update uk

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update mk

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update es

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update ja

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update fa

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update pl

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update sv

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update ko

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update sr

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update nl

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update tr

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update be

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update it

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update id

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update ms

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update bg

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update az

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update en

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update ru

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update pt-pt

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update de

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update zh-tw

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update cs

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update pt-br

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update sl

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update fr

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update zh

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update uz

Updated via the `update-book.yml` GitHub workflow.

Commit:
book: update tl

Updated via the `update-book.yml` GitHub workflow.

Commit:
Merge branch 'book-bg' of bundle-bg/bg.bundle into gh-pages

Commit:
Merge branch 'book-be' of bundle-be/be.bundle into gh-pages

Commit:
Merge branch 'book-cs' of bundle-cs/cs.bundle into gh-pages

Commit:
Merge branch 'book-en' of bundle-en/en.bundle into gh-pages

Commit:
Merge branch 'book-de' of bundle-de/de.bundle into gh-pages

Commit:
Merge branch 'book-fr' of bundle-fr/fr.bundle into gh-pages

Commit:
Merge branch 'book-es' of bundle-es/es.bundle into gh-pages

Commit:
Merge branch 'book-id' of bundle-id/id.bundle into gh-pages

Commit:
Merge branch 'book-gr' of bundle-gr/gr.bundle into gh-pages

Commit:
Merge branch 'book-ja' of bundle-ja/ja.bundle into gh-pages

Commit:
Merge branch 'book-it' of bundle-it/it.bundle into gh-pages

Commit:
Merge branch 'book-ko' of bundle-ko/ko.bundle into gh-pages

Commit:
Merge branch 'book-ms' of bundle-ms/ms.bundle into gh-pages

Commit:
Merge branch 'book-mk' of bundle-mk/mk.bundle into gh-pages

Commit:
Merge branch 'book-nl' of bundle-nl/nl.bundle into gh-pages

Commit:
Merge branch 'book-pl' of bundle-pl/pl.bundle into gh-pages

Commit:
Merge branch 'book-pt-pt' of bundle-pt-pt/pt-pt.bundle into gh-pages

Commit:
Merge branch 'book-pt-br' of bundle-pt-br/pt-br.bundle into gh-pages

Commit:
Merge branch 'book-ru' of bundle-ru/ru.bundle into gh-pages

Commit:
Merge branch 'book-sr' of bundle-sr/sr.bundle into gh-pages

Commit:
Merge branch 'book-sl' of bundle-sl/sl.bundle into gh-pages

Commit:
Merge branch 'book-tr' of bundle-tr/tr.bundle into gh-pages

Commit:
Merge branch 'book-tl' of bundle-tl/tl.bundle into gh-pages

Commit:
Merge branch 'book-uz' of bundle-uz/uz.bundle into gh-pages

Commit:
Merge branch 'book-uk' of bundle-uk/uk.bundle into gh-pages

Commit:
Merge branch 'book-sv' of bundle-sv/sv.bundle into gh-pages

Commit:
Merge branch 'book-zh' of bundle-zh/zh.bundle into gh-pages

Commit:
Merge branch 'book-zh-tw' of bundle-zh-tw/zh-tw.bundle into gh-pages

Commit:
Merge branch 'book-fa' of bundle-fa/fa.bundle into gh-pages

jvns commented:
No worries, it was easy to resolve!

To1ne commented:
Boom! :boom:
Thanks!
